src/App.tsx

```diff
@@ -1,26 +1,393 @@
+        .finally(() => {
+          setTempTodo(null);
+        });
+    }
+  }
+
+  function onDelete(id: number): Promise<void> {
+    setIsDeletingTodo(true);
+
+    return client
+      .delete(`/todos/${id}`)
+      .then(() => {
+        setTodos(prevTodos => removeTodoById(prevTodos, id));
+        setFilteredTodos(prevTodos => removeTodoById(prevTodos, id));
+      })
+      .catch(() => ShowError(Error.DeleteError))
+      .finally(() => {
+        setIsDeletingTodo(false);
+      });
+  }
+
+  function handleClearCompleted() {
+    const completedTodoIds = todos.reduce(
+      (acc, todo) => (todo.completed ? [...acc, todo.id] : acc),
+      [] as number[],
+    );
+
+    setTodoToDeleteIds(completedTodoIds);
+
+    const promises: Promise<void>[] = [];
+
+    completedTodoIds.forEach(id => {
+      promises.push(onDelete(id));
+    });
+
+    Promise.all(promises).then(() => {
+      //винести
+      if (addTodoField.current !== null) {
+        addTodoField.current.focus();
+      }
+    });
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {/* this button should have `active` class only if all todos are completed +++*/}
+          <button
+            type="button"
+            className={cN('todoapp__toggle-all', { active: isAllCompleted })}
+            data-cy="ToggleAllButton"
+          />
+
+          {/* Add a todo on form submit -*/}
+          <form onSubmit={handleAddTodoOnEnter}>
+            <input
+              ref={addTodoField}
+              data-cy="NewTodoField"
+              type="text"
+              value={inputText}
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              onChange={event => setInputText(event.target.value)}
+              disabled={tempTodo !== null}
+            />
+          </form>
+        </header>
+        <section className="todoapp__main" data-cy="TodoList">
+          {filteredTodos.map(todo => (
+            <TodoItem
+              key={todo.id}
+              todo={todo}
+              handleTodoStatusChange={handleTodoStatusChange}
+              onDelete={onDelete}
+              isDeletingTodo={isDeletingTodo}
+              todoToDeleteIds={todoToDeleteIds}
+              setTodoToDeleteIds={setTodoToDeleteIds}
+              addTodoField={addTodoField}
+            />
+          ))}
+          {tempTodo !== null && (
+            <TodoItem key={tempTodo.id} todo={tempTodo} isTemp={true} />
```

Comment:
1) Don't forget about indents
2) Use `!!` to convert ur value to the boolean type

```suggestion
          </form>
        </header>
        
        <section className="todoapp__main" data-cy="TodoList">
          {filteredTodos.map(todo => (
            <TodoItem
              key={todo.id}
              todo={todo}
              handleTodoStatusChange={handleTodoStatusChange}
              onDelete={onDelete}
              isDeletingTodo={isDeletingTodo}
              todoToDeleteIds={todoToDeleteIds}
              setTodoToDeleteIds={setTodoToDeleteIds}
              addTodoField={addTodoField}
            />
          ))}
          
          {!!tempTodo && (
            <TodoItem key={tempTodo.id} todo={tempTodo} isTemp={true} />
```